js/initcontext.go

```diff
@@ -140,11 +140,11 @@ func (i *InitContext) Require(arg string) goja.Value {
 }
 
 func (i *InitContext) requireModule(name string) (goja.Value, error) {
-	mod := modules.Get(name)
-	if mod == nil {
+	mod, ok := i.modules[name]
+	if !ok {
 		return nil, errors.Errorf("unknown module: %s", name)
 	}
-	return i.runtime.ToValue(common.Bind(i.runtime, mod, i.ctxPtr)), nil
+	return i.runtime.ToValue(common.Bind(i.runtime, mod.NewModuleInstancePerVU(), i.ctxPtr)), nil
```

Comment:
Connected to my other comment, what's the problem of keeping this logic here?

```suggestion
	if i, ok := mod.(HasModuleInstancePerVU); ok {
		mod = i.NewModuleInstancePerVU()
	}

	return i.runtime.ToValue(common.Bind(i.runtime, mod, i.ctxPtr)), nil
```